x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/ilm/IndexLifecycleServiceTests.java

```diff
@@ -285,10 +285,7 @@ private void verifyCanStopWithStep(String stoppableStep) {
             changedOperationMode.set(true);
             return null;
         }).when(clusterService)
-            .submitUnbatchedStateUpdateTask(
-                eq("ilm_operation_mode_update[stopped]"),
-                eq(OperationModeUpdateTask.ilmMode(OperationMode.STOPPED))
```

Comment:
[Thinking out loud] If I am not mistaken, now the test is a bit weaker than before, we do not verify anymore that the mode in `OperationModeUpdateTask` is `STOPPED`. 

I still think removing the equals is a good idea but maybe we can think of an alternative way to test that the operation mode that was given as an argument is correct. For example, we could inspect the argument via `invocationOnMock.getArguments()`.

Reply:
Yeah I thought about that as well. The reason I still did it this way was that we do the same [here](https://github.com/nielsbauman/elasticsearch/blob/874d0311d034420828ae06d2df58306cdcc6bb66/x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/ilm/IndexLifecycleServiceTests.java#L340-L340). I initially didn't dive into the functionality itself to check whether we actually need to check the mode of `OperationModeUpdateTask` in that parameter.

However, now that I look at the code, it looks like [this](https://github.com/nielsbauman/elasticsearch/blob/874d0311d034420828ae06d2df58306cdcc6bb66/x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IndexLifecycleService.java#L251-L251) is the only place we call `submitUnbatchedStateUpdateTask` so I'd argue that it doesn't really make sense to check for the mode in this case. What do you think?

Reply:
> Yeah I thought about that as well. The reason I still did it this way was that we do the same [here](https://github.com/nielsbauman/elasticsearch/blob/874d0311d034420828ae06d2df58306cdcc6bb66/x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/ilm/IndexLifecycleServiceTests.java#L340-L340). I initially didn't dive into the functionality itself to check whether we actually need to check the mode of `OperationModeUpdateTask` in that parameter.

I would say that what we do there is indeed correct, both tests are checking if the ILM stopping task is submitted successfully. So, checking that the task is a `STOPPED` task, I would say is a core part of the test.

Consider it this way, in the new version, this test checks that this path of the code submits any `OperationModeUpdateTask` and not necessarily a `STOPPED` one, so the assertion does not satisfy what the name of the test says it's testing `verifyCanStopWithStep`.

> However, now that I look at the code, it looks like [this](https://github.com/nielsbauman/elasticsearch/blob/874d0311d034420828ae06d2df58306cdcc6bb66/x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IndexLifecycleService.java#L251-L251) is the only place we call `submitUnbatchedStateUpdateTask` so I'd argue that it doesn't really make sense to check for the mode in this case. What do you think?

I see your point, you can see from the code how it's called so you know that it's already covered. But think about this, [stopILM](https://github.com/nielsbauman/elasticsearch/blob/874d0311d034420828ae06d2df58306cdcc6bb66/x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IndexLifecycleService.java#L251-L251) is a private method. It could change at any moment and this test will not be able to catch a bug that will be introduced in the future. 

Do the above make sense?

Reply:
Ah my bad, I missed [this](https://github.com/nielsbauman/elasticsearch/blob/e1f2a6a101890163f8bf659fb953d39540b5682d/x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/ilm/IndexLifecycleServiceTests.java#L336-L336) line. Now I see that they do that check in the other test as well :).

Will update this test as well, thanks!